Gemfile

```diff
@@ -5,3 +5,6 @@ gem "rugged"
 gem "xcodeproj", "~> 1.3.3"
 gem "nokogiri"
 gem "activesupport", [">= 4.2", "< 5.0"] # 5.0 requires Ruby 2.2.2 so force an older version
+
+custom_gemfile_path = File.join(File.dirname(__FILE__), 'custom', 'Gemfile')
+eval(File.read(custom_gemfile_path), binding) if File.exist?(custom_gemfile_path)
```

Comment:
I've tried it.
However, it won't work.
`require_relative` behaves like `require`, so passing path must be not for `Gemfile` but Ruby modules.

I copied this code from Fastlane's plugin system, so it might be correct.